### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by Hombre2014**
All of the requirements for this phase of the project have been implemented:

### Project requirements
-  Use Capybara to write integration tests for each view in your project.

- Login page:
  - I can see the username and password inputs and the "Submit" button.
  - When I click the submit button without filling in the username and the password, I get a detailed error.
  - When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
  - When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

**Be sure to include integration specs for any other features you may have implemented!**

### CODE REVIEW

File: spec/pages/post_index_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      Post.create!(
+        author: user,
+        title: 'First title',
+        text: 'First text',
+        likes_counter: 0,
+        comments_counter: 0
+      ),
+      Post.create!(
+        author: user,
+        title: 'Second title',
```
Comment (andersonlebon) [inline spec/pages/post_index_page_spec.rb]: - [ ] Kindly make sure all of your tests are passing. these are falling tests.

image

- [ ] Also you need kindly provide instructions on running your test ass your tests were only running after using these commands: `bin/rails generate rspec:install`, `bundle exec rspec`.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Hombre2014 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/pages/post_index_page_spec.rb (hunk 1 above)
Comment (Hombre2014) [reply spec/pages/post_index_page_spec.rb]: Thanks for your recommendations. Somehow when you run all the tests with bundle exec rspec the tests failed. Please, run all the tests individually like:

`bundle exec rspec spec/pages/login_page_spec.rb`
`bundle exec rspec spec/pages/post_idex_page_spec.rb`
`bundle exec rspec spec/pages/post_show_page_spec.rb`
`bundle exec rspec spec/pages/user_index_page_spec.rb`
`bundle exec rspec spec/pages/user_show_page_spec.rb`

and they will passed.

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
### When I tested your project as a whole group I got 24 errors out of 54. The reason why I am accepting this project is that all the tests did pass when I was testing them individually(by page). But I would recommend you take a step back and check for the reason for this happening.
Screenshot from 2022-04-14 23-27-44

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Hombre2014** [comment]
Thank you very much for your consideration!

**Pull request merged**